pr47-core/src/data/exception.rs

```diff
@@ -7,7 +7,9 @@ pub enum UncheckedException {
     InvalidBinaryOp { bin_op: char, lhs: Value, rhs: Value, reason: String }
 }
 
+pub type CheckedException = Box<dyn 'static + Error + Send>;
```

Comment:
If an exception is going to be used in by Pr47 program, then it must satisfy `StaticBase` somehow, and can be type-checked and casted at run time. I suggest a Box<dyn DynBase>.

Reply:
**[MID]** Then, here comes the question: for the `Error`s that do not implement `StaticBase`, user may hope them to be `UncheckedException`s. That makes sense I think. However we don't have `specialisation` or `min-specialisation` yet.